error.go

```diff
@@ -0,0 +1,43 @@
+package consul
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// IsNotFound reports whether the error is a not found error.
+func IsNotFound(err error) bool {
```

Comment:
Is it unnecessary for this function to be part of the consul-go public API?

Reply:
should I expose that, the error, or an interface ?

Reply:
Or nothing? There's no need to extend the public API of the package for the dynamic error type checking to work. We just have to document the behavior.